haas/api.py

```diff
@@ -274,7 +327,7 @@ def _have_attachment(nic, query):
     if nic.current_action:
         raise BlockedError("A networking operation is already active on the nic.")
 
-    if (network.access is not None) and (network.access is not project):
+    if (network.access) and (project not in network.access):
```

Comment:
It errors out

if I'm remembering right it's because network access is sometimes None. It might make sense to change the way  network access is handled for public networks.